server/src/main/java/org/elasticsearch/repositories/RepositoryData.java

```diff
@@ -816,10 +824,29 @@ public static RepositoryData snapshotsFromXContent(XContentParser parser, long g
                     indexMetaIdentifiers = parser.mapStrings();
                 }
                 case MIN_VERSION -> {
-                    XContentParserUtils.ensureExpectedToken(XContentParser.Token.VALUE_STRING, parser.nextToken(), parser);
-                    final Version version = Version.fromString(parser.text());
+                    IndexVersion version;
+                    switch (parser.nextToken()) {
+                        case VALUE_STRING -> {
+                            Version v = Version.fromString(parser.text());
+                            assert v.before(Version.V_8_10_0);
+                            version = IndexVersion.fromId(v.id);
+                        }
+                        case VALUE_NUMBER -> {
+                            version = IndexVersion.fromId(parser.intValue());
+                        }
+                        default -> throw new ParsingException(
```

Comment:
There's a bug here, you would want to pass `parser.currentToken()` to the string formatting below, but that said:
I think we can just use `org.elasticsearch.common.xcontent.XContentParserUtils#throwUnknownToken` here, it gives us the same amount of information without blowing up the code size so much.

Reply:
Ah yes, didn't know about that method!